File: 3931-check-adjacent-digit-differences/3931-check-adjacent-digit-differences.ex
defmodule Solution do
  @spec is_adjacent_diff_at_most_two(s :: String.t()) :: boolean
  def is_adjacent_diff_at_most_two(s) do
    s
    |> String.to_charlist()
    |> Enum.chunk_every(2, 1, :discard)
    |> Enum.all?(fn [a, b] -> abs(a - b) <= 2 end)
  end
end

File: 3931-check-adjacent-digit-differences/3931-check-adjacent-digit-differences.php
class Solution {

    /**
     * @param String $s
     * @return Boolean
     */
    function isAdjacentDiffAtMostTwo($s) {
        for ($i = 1; $i < strlen($s); $i++) {
            if (abs(ord($s[$i]) - ord($s[$i - 1])) > 2) {
                return false;
            }
        }
        return true;
    }
}